packages/core/src/raw.ts

```diff
@@ -10,23 +10,37 @@ export type Sql = {
   values: unknown[];
 };
 
-export type RawExpression<C extends ColumnTypeBase = ColumnTypeBase> = {
-  __raw: string | TemplateLiteralArgs;
-  __values?: Record<string, unknown> | false;
-  __column: C;
-};
+export class RawExpression<C extends ColumnTypeBase = ColumnTypeBase> {
+  constructor(
+    readonly __raw: string | TemplateLiteralArgs,
+    readonly __values?: Record<string, unknown> | false,
+    readonly __column: C = undefined as unknown as C,
+  ) {}
+
+  /**
+   * Static-cast untyped raw expression to the provided type.
+   *
+   * ```ts
+   * db.sql`data->>'field'`.castTo<string>()
+   * ```
+   */
+  castTo<T extends C['type']>(): RawExpression<
+    ColumnTypeBase<T, C['operators'], C['inputType'], C['data']>
+  > {
+    return this as RawExpression<
+      ColumnTypeBase<T, C['operators'], C['inputType'], C['data']>
+    >;
+  }
+}
 
 export const raw = <C extends ColumnTypeBase = ColumnTypeBase>(
```

Comment:
Could you rename `raw` to `sql`, along the way? Not a blocker, can be renamed later.